core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -89,12 +89,14 @@ object ZkUtils {
   }
 
   def maybeDeletePath(zkUrl: String, dir: String) {
+    val zk = createZkClient(zkUrl, 30*1000, 30*1000)
     try {
-      val zk = createZkClient(zkUrl, 30*1000, 30*1000)
       zk.deleteRecursive(dir)
-      zk.close()
     } catch {
+      case _: ZkException => zk.deleteRecursive(dir)
```

Comment:
Unfortunate that we cannot catch a more specific error here. Perhaps at least we can add a comment explaining the case that we're trying to handle?